### ISSUE TITLE
Remove cache dir/file on postinst, not preinst

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-client#352]
We should remove the cache dir and file, from old ua, in postinst, not preinst:
```
<vorlon> andreas: I don't see any reason for the preinst bits to happen in preinst instead of postinst, can you explain?
<andreas> vorlon: without that, dpkg was issuing a warning that it couldn't remove a non-empty directory
<vorlon> oh
<andreas> when upgrading from a package that had the cache file in there, i.e., cron.daily ran at least once in the past
<vorlon> those warnings are no big deal as long as your maintainer script takes care of it afterwards
<andreas> ok, but by removing the file, and leaving the directory, dpkg removed the (now empty) directory in the end
<vorlon> so I would remove the cache file and the cache dir in the postinst, not in the preinst, because preinst scripts make apt/dpkg more rittle
<vorlon> brittle
<vorlon> (this is also not a blocker)
```

**Comment by powersj** [comment]
Can you confirm this is for the old ua client? Was this done?

I'm inclined not to do this unless we absolutely must since we don't really want to be uploading the old client.

**Comment by panlinux** [comment]
This is for the new client that was uploaded to disco.

**Comment by panlinux** [comment]
By moving it to postinst, we get the dpkg warning as expected:
```
$ sudo dpkg -i ../*.deb
(Reading database ... 95947 files and directories currently installed.)
Preparing to unpack .../ubuntu-advantage-tools_19.5_amd64.deb ...
Unpacking ubuntu-advantage-tools (19.5) over (17) ...
dpkg: warning: unable to delete old directory '/var/cache/ubuntu-advantage-tools': Directory not empty
Setting up ubuntu-advantage-tools (19.5) ...
Removing obsolete conffile /etc/update-motd.d/80-esm ...
Removing obsolete conffile /etc/update-motd.d/80-livepatch ...
Removing obsolete conffile /etc/cron.daily/ubuntu-advantage-tools ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
```
But the directory is gone in the end:
```$ l /var/cache/ubuntu-advantage*
ls: cannot access '/var/cache/ubuntu-advantage*': No such file or directory
```

**Comment by powersj** [comment]
Closed due to 14ad98d
